README.md

```diff
@@ -3,3 +3,17 @@
 # `lnschema_core`
 
 `lnschema_core` is documented & tested within `lamindb`: [github.com/laminlabs/lamindb](https://github.com/laminlabs/lamindb).
+
+## Developers' Note
+
+This package implements models but leaves concrete implementations of non-ORM methods (such as `Artifact.cache`) empty. The methods are implemented in `lamindb` and then the model classes are monkey-patched there as well.
+
+### Why monkey pathing?
+
+We want to avoid implementing anything non-ORM related in this package and just have models here, so that concrete implementations are all in `lamindb`. This provides better code structure and clarity compared to having imports from `lamindb` needed if the implementations are done here.
+
+### Why not simple inheritance?
+
+It is possible to replace monkey patching with simple inheritance for some model classes (like `Artifact`), but it is hard to do for all due to the situation when a model class inherits another model class within `lnschema-core`. This is true, for example, for `Artifact` inheriting `Record`.
```

Comment:
It is **not** true for `Artifact` due to its inheritance from `Record`, right? I'm confused reading this.

Reply:
If record is monkey-patched, it is easy to replace monkey-patching of `Artifact` with simple inheritance.

Reply:
And why don't we want that? 🤔

Reply:
I think we discussed this. You said you didn't want also mixing of monkey patching and simple inheritance.

Reply:
Ok! 🤔